mediaengine.go

```diff
@@ -0,0 +1,141 @@
+package webrtc
+
+import (
+	"github.com/pions/webrtc/pkg/rtp"
+	"github.com/pions/webrtc/pkg/rtp/codecs"
+)
+
+var rtcMediaEngine = &mediaEngine{}
+
+func init() {
+	RegisterCodec(NewRTCRtpOpusCodec(PayloadTypeOpus, 48000, 2))
+	RegisterCodec(NewRTCRtpVP8Codec(PayloadTypeVP8, 90000))
+	RegisterCodec(NewRTCRtpH264Codec(PayloadTypeH264, 90000))
+}
+
+// RegisterCodec allows an implementation to provide it's own codec.
+func RegisterCodec(codec *RTCRtpCodec) {
+	rtcMediaEngine.RegisterCodec(codec)
+}
+
+func (m *mediaEngine) ClearCodecs() {
+	rtcMediaEngine.ClearCodecs()
+}
+
+type mediaEngine struct {
+	codecs []*RTCRtpCodec
+}
+
+func (m *mediaEngine) RegisterCodec(codec *RTCRtpCodec) {
+	m.codecs = append(m.codecs, codec)
+}
+
+func (m *mediaEngine) ClearCodecs() {
+	m.Codecs = nil
+}
+
+// NewRTCRtpOpusCodec is a helper to create an Opus codec
+func NewRTCRtpOpusCodec(payloadType int, clockrate UnsignedLong, channels UnsignedShort) *RTCRtpCodec {
+	NewRTCRtpCodec(RTCRtpCodecKindAudio,
+		"opus",
+		clockrate,
+		channels,
+		"minptime=10;useinbandfec=1",
+		payloadType,
+		&codecs.OpusPayloader{})
+}
+
+// NewRTCRtpVP8Codec is a helper to create an VP8 codec
+func NewRTCRtpVP8Codec(payloadType int, clockrate UnsignedLong) *RTCRtpCodec {
+	NewRTCRtpCodec(RTCRtpCodecTypeVideo,
+		"VP8",
+		clockrate,
+		0,
+		"",
+		payloadType,
+		&codecs.VP8Payloader{})
+}
+
+// NewRTCRtpH264Codec is a helper to create an VP8 codec
+func NewRTCRtpH264Codec(payloadType int, clockrate UnsignedLong) *RTCRtpCodec {
+	NewRTCRtpCodec(RTCRtpCodecTypeVideo,
+		"H264",
+		clockrate,
+		0,
+		"",
+		payloadType,
+		&codecs.VP8Payloader{})
```

Comment:
H264

Reply:
Fixed typo. Hiding comment since it's on another line.